### DISCUSSION TITLE
Questions regarding the ZK-Desktop view

### DISCUSSION THREAD

**Discussion by anthrolisp** [localauthor/zk#58] [Ideas]
I can imagine the ZK-Desktop view as a place to work on different writing projects. Two thoughts:
- It would be interesting to be able to export a ZK-Desktop view along with linked notes (embedded/transcluded would be interesting!)
- and/or be able to concatenate the contents of a list of files (or file links) to a separate file (possibly with some kind of reference inside that file to where the contents came from)

**Comment by localauthor** [comment]
Interesting ideas! The following comes to mind, just as a starting point --- trying running this in a buffer with zk-ids:

``` elisp
(defun zk-concat ()
  (interactive)
  (goto-char (point-min))
  (while (re-search-forward zk-id-regexp nil nil)
    (let ((id (match-string-no-properties 1)))
      (get-buffer-create "ZK-Concat")
      (with-current-buffer "ZK-Concat"
        (insert-file (zk--parse-id 'file-path id))
        (newline)))))
```
